cirq-ft/cirq_ft/algos/select_and_prepare.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2023 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+
+from cirq._compat import cached_property
+from cirq_ft import infra
+
+
+class SelectOracle(infra.GateWithRegisters):
```

Comment:
Should these two interfaces have docstrings to explain the purpose of the interface and how/if to subclass them?

Reply:
added